### ISSUE TITLE
color settings is not working

### ISSUE THREAD

**Issue by rashidnkx-novigo** [oreporan/color-tabs-vscode#39]
I have installed extension  and copied settings to .vscode/settings.json

```
    "colorTabs.config": [
        {
            "regex": ".*\/ui\/.*",
            "color": "#FF0000",
            "label": "MOBILE"
        },
        {
            "regex": ".*\/app\/.*",
            "color": "#00FF00",
        },
        {
            "regex": ".*\/desktop\/.*"
        },
    ],
    "colorTabs.titleBackground": true,
    "colorTabs.ignoreCase": true
```

I don't see any changes when I select file from both directories

**Comment by oreporan** [comment]
Usually its some regex error
Can you paste some file directory path you're testing on?

**Comment by rashnk** [comment]
I have two folders on the project root, ui and app can you help me to set the regex for this

**Comment by oreporan** [comment]
yeah thats why I asked for you to paste a file for example so we can see why it doesn't work.
Also are you running linux/mac or windows?

**Comment by oreporan** [comment]
closing this

**Comment by gusbemacbe** [comment]
I can confirm that it does not work. I applied to the user (not workspace) `settings.json`. I am running on Linux, @oreporan.

**Comment by oreporan** [comment]
Hey @gusbemacbe 
Please provide the config you set in your settings.json, and some file you've tested it on so we can see

**Comment by gusbemacbe** [comment]
Hello @oreporan !

Only the regex `css` toggled the colour, but other regexes did not toggle. I recorded a video. You can check the video:

https://giant.gfycat.com/DependentThriftyHawaiianmonkseal.webm

```json
{
    "colorTabs.config": 
    [
        {
            "regex": ".*\/config\/.*",
            "color": "#00FF00",
        },
        {
            "regex": ".*\/css\/.*",
            "color": "#ABCDEF",
        },
        {
            "regex": ".*\/settings\/.*",
            "color": "#b03ea8",
        }
    ],
}
```

**Comment by gusbemacbe** [comment]
Your extension confounded `config` with `settings`, and toggled `config`'s colour to `settings`:

![image](https://user-images.githubusercontent.com/2257024/90747687-a1314d00-e2a7-11ea-8760-f34ac7607cf4.png)

**Comment by oreporan** [comment]
Look at the path of your settings folder, its:

`home/gusbemacbe/.config/.../settings.json`

so the word `config` is part of the path. 
I'm guessing if you change the order of your ColorTabs regex configuration it will work

**Comment by gusbemacbe** [comment]
As you have suggested, I reordered, now the `config` and `settings` worked, but `css` stopped working. I tested `yaml`, and it did not toggle. 

```json
    "colorTabs.config": 
    [
        {
            "regex": "^config*",
            "color": "#00FF00",
        },
        {
            "regex": "^(settings)*",
            "color": "#b03ea8",
        },
        {
            "regex": ".*\/css\/.*",
            "color": "#ABCDEF",
        },
        {
            "regex": ".*\/yaml\/.*",
            "color": "#ebbd5b",
        }
    ]
```

**Comment by oreporan** [comment]
What are you trying to catch with  `"regex": ".*\/css\/.*"`
This regex will catch a file in this format: 

`x/css/y`

Use some online [regex tester online](https://regexr.com/) if you don't know how to write regular expressions

**Comment by gusbemacbe** [comment]
I tried the own regexes, and all other regexes I found in the community patterns, but nothing worked.

**Comment by oreporan** [comment]
The regex you showed me `"regex": ".*\/css\/.*",` will not catch files with a `.css` ending if thats what you wanted.

Are you trying to do this?

```json
  "colorTabs.config": [{
        "regex": ".*.css",
        "color": "#FF0000",
    }]
```

**Comment by gusbemacbe** [comment]
> Are you trying to do this?

Yes, I did, but unluckily it did not toggle the colour. I began to understand your extension supports only directories and not file extensions/formats

**Comment by oreporan** [comment]
Not true, it does indeed support file extensions. 

The regex I wrote above worked for me:

```json
  "colorTabs.config": [{
        "regex": ".*.css",
        "color": "#FF0000",
    }]
```

Did you make sure it doesn't collide with another ColorTabs regex configuration above it?

Try and remove all configurations except your CSS configuration

**Comment by gusbemacbe** [comment]
> Did you make sure it doesn't collide with another ColorTabs regex configuration above it?
> 
> Try and remove all configurations except your CSS configuration

Ah, I finally discovered a bug. It is order problem. See the screenshot. I had to comment the first two regexes in order, then CSS regex worked now. 

If I uncomment these two regexes and keep that order, CSS regex will stop working. 

![image](https://user-images.githubusercontent.com/2257024/90777997-7c060400-e2d2-11ea-8b0e-9fb51cb138f0.png)
![image](https://user-images.githubusercontent.com/2257024/90778063-904a0100-e2d2-11ea-8d2e-e742431ab8c7.png)

And YAML regex also worked:

![image](https://user-images.githubusercontent.com/2257024/90778167-b40d4700-e2d2-11ea-979d-f6ef8967d2f2.png)
